File: data/imported/knowledge/cells/cpp/merged/snippets/snippet_024.cpp
TEST(CellDataGoldenTests, shared_strings_export_uses_sst_when_enabled)
{
    TempDir temp("golden-sst");
    const auto path = temp.Path("shared-strings.xlsx");
    Workbook workbook;
    workbook.GetWorksheets()[0].GetCells()["A1"].PutValue("Hello");
    workbook.GetWorksheets()[0].GetCells()["A2"].PutValue("Hello");
    SaveOptions options;
    options.SetUseSharedStrings(true);
    workbook.Save(path.string(), options);

    EXPECT_TRUE(Package::EntryExists(path, "xl/sharedStrings.xml"));
    const auto sharedStringsXml = Package::ReadEntryText(path, "xl/sharedStrings.xml");
    const auto worksheetXml = Package::ReadEntryText(path, "xl/worksheets/sheet1.xml");

    EXPECT_TRUE(Contains(sharedStringsXml, "<sst"));
    EXPECT_TRUE(Contains(sharedStringsXml, "uniqueCount=\"1\""));
    EXPECT_TRUE(Contains(worksheetXml, "t=\"s\""));
}

File: data/imported/knowledge/cells/typescript/merged/snippets/snippet_024.ts
async function testWorksheetAccess() {
  const workbook = new Workbook();

  workbook.worksheets.addWorksheet("FirstSheet");
  workbook.worksheets.addWorksheet("SecondSheet");
  workbook.worksheets.addWorksheet("ThirdSheet");

  console.log("Worksheet 0:", workbook.worksheets[0]?.name);
  console.log("Worksheet 1:", workbook.worksheets[1]?.name);
  console.log("Worksheet 2:", workbook.worksheets[2]?.name);

  await workbook.save("outputfiles/test_worksheet_access.xlsx");
}